src/stores/AccountController.ts

```diff
@@ -13,22 +25,11 @@ import { restoreWatchHistory, serializeWatchHistory } from '#src/stores/WatchHis
 import { useWatchHistoryStore } from '#src/stores/WatchHistoryStore';
 import { logDev } from '#src/utils/common';
 import * as persist from '#src/utils/persist';
-import type {
-  Capture,
-  Customer,
-  CustomerConsent,
-  EmailConfirmPasswordInput,
-  FirstLastNameInput,
-  GetCaptureStatusResponse,
-  GetCustomerConsentsResponse,
-  GetPublisherConsentsResponse,
-} from '#types/account';
 import type { Offer } from '#types/checkout';
-import { unpersistProfile } from '#src/hooks/useProfiles';
 
 const PERSIST_PROFILE = 'profile';
 
-export const initializeAccount = async () => {
+export const initializeAccount = async ({ profile }: { profile?: Profile } = {}) => {
```

Comment:
The initialize account function intends to initialize all account-related stores and services only once. Service could bind event listeners or initialize actions that shouldn't be performed multiple times.

It is entangled with profile logic and called when switching the profile.

Would something like this work?

```mermaid
---
title: Account/Profile service flow
---
flowchart TD
    START[/Startup/] --> CONFIG(Load config)
    CONFIG --> INITP[Profile.initialize]
    INITP --> INITA[Account.initialize]
    INITA --> LOADU[Account.loadUser]
    LOADU --> EMPTY[[Empty favorites/history]]
    EMPTY --> AUTH[[getAuthData]]
    AUTH --> END[[account/watchHistory/favorites]]
    SELECT[/Select profile/] --> SWITCH[Profile.switchProfile]
    SWITCH --> PERSIST[Profile.persistProfile]
    PERSIST --> LOADU
```

Reply:
You're right, there will be constant unnecessary calls to initializeAccount. I've separated the logic a bit here: https://github.com/jwplayer/ott-web-app/pull/371/commits/d358ddbf0613ff5e0d0bb8e483770219761cec8d

Not exactly like your example but I think it should achieve the same result.

Reply:
I've also tried to address the rest of the comments in the same commit